overlord/hookstate/hookmgr.go

```diff
@@ -336,13 +350,9 @@ func (m *HookManager) runHook(context *Context, snapst *snapstate.SnapState, hoo
 
 	if hookExists || mustHijack {
 		// we will run something, not a noop
-		if ok, _ := m.state.Restarting(); ok {
-			// don't start running a hook if we are restarting
-			return &state.Retry{}
+		if err := m.runGuard(context); err != nil {
```

Comment:
(nitpick) maybe a comment here that `runGuard` checks for restarting or something? The name is quite generic so a bit more context might be helpful for the causal reader :) (but I tried to come up with a more expressive name for the func and couldn't so maybe it's fine).

Reply:
runHookRestartingGuard ?